### ISSUE TITLE
Should return_prompt always be False for training a decoder model?

### ISSUE THREAD

**Issue by dkkim93** [lvwerra/trl#497]
Hello! Thank you all for this fantastic repository 👍 I have loved using trl :)

While looking at the code more closely, the [generate](https://github.com/lvwerra/trl/blob/2b531b92238f9fc40c6cc9cf27b8bbd3ae2727b1/trl/trainer/ppo_trainer.py#L407C11-L407C12) function has the `return_prompt` variable set to be `True` as default. 
However, the [prepare_model_inputs](https://github.com/lvwerra/trl/blob/2b531b92238f9fc40c6cc9cf27b8bbd3ae2727b1/trl/trainer/ppo_trainer.py#L808) function inside the [ppo_trainer.step](https://github.com/lvwerra/trl/blob/2b531b92238f9fc40c6cc9cf27b8bbd3ae2727b1/trl/trainer/ppo_trainer.py#L596C6-L596C8) function concats the query and response tensors: `input_ids = [torch.cat([q, r]) for q, r in zip(queries, responses)]` [[link]](https://github.com/lvwerra/trl/blob/2b531b92238f9fc40c6cc9cf27b8bbd3ae2727b1/trl/trainer/ppo_trainer.py#L822C13-L822C80) for the decoder only model. 

If `return_prompt=False`, then the concatenation perfectly makes sense. However, if `return_prompt=True`, I wonder whether the concatenation can cause undesirable training performance :thinking: and whether we should put some warning flag. 

Thank you again for this awesome code!

**Comment by lvwerra** [comment]
We always pass `query`/`response` separately to to `PPOTrainer.step()` so that we know which part to mask. When you generate you already have the `query` and when you set `return_prompt=False` then you also only get the `response` rather than needing to split `query+response` again. So mainly a quality of life option.

**Comment by dkkim93** [comment]
This makes sense. Thank you for your clarification!
